### PR TITLE
[ADD] 17.0 analysis

### PR DESCRIPTION
**Pull request by StefanRijnhart**

### CODE REVIEW

File: openupgrade_scripts/readme/DESCRIPTION.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+This module is a containers of migration script to migrate from 15.0 to
+16.0 version.
```
Comment (legalsylvain) [inline openupgrade_scripts/readme/DESCRIPTION.md]:
```suggestion
This module is a containers of migration script to migrate from 16.0 to
17.0 version.
```

File: requirements.txt [hunk 2]
```diff
@@ -1 +1,2 @@
-openupgradelib @ git+https://github.com/OCA/openupgradelib@master
+# generated from manifests external_dependencies
+openupgradelib
```
Comment (legalsylvain) [inline requirements.txt]: Why this change ? 
As soon as merging a PR in openupgradelib doens't generate a new package on pypi, I guess that we should keep the url "openupgradelib @ git+https://github.com/OCA/openupgradelib@master"

Don't you think ?

**Review by legalsylvain — COMMENTED** [review]
Minor remarks inline. 

Thanks for initializing OpenUpgrade 17.0 !

File: openupgrade_scripts/apriori.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+renamed_models = {
+    # odoo
+    # OCA/...
+}
+
+# only used here for upgrade_analysis
+merged_models = {
+    # odoo
+    # OCA/...
+}
```
Comment (legalsylvain) [inline openupgrade_scripts/apriori.py:50]: This file is empty. FYI, I began a list of renamed / merged Odoo modules here : 
https://github.com/OCA/OpenUpgrade/issues/4124#issue-1888214543

File: openupgrade_scripts/readme/DESCRIPTION.md (hunk 1 above)
Comment (StefanRijnhart) [reply openupgrade_scripts/readme/DESCRIPTION.md]: Thanks, fixed now!

File: requirements.txt (hunk 2 above)
Comment (StefanRijnhart) [reply requirements.txt]: Thanks! Fixed by setting the external dependency in the module manifest to this git notation.

File: openupgrade_scripts/apriori.py (hunk 3 above)
Comment (StefanRijnhart) [reply openupgrade_scripts/apriori.py:50]: Thanks! Great work on the renamed and merge modules. I updated apriori with it and reran the analysis.

**Comment by alexisadrianc** [comment]
:+1:

**Review by legalsylvain — APPROVED** [review]
Thanks ! 

/ocabot merge nobump

**Comment by legalsylvain** [comment]
@sbidoul : maybe do you have an idea about the correct syntax to add git dependency for python package. 
here is the current code : 
```
    "external_dependencies": {
        "python": [
            "openupgradelib @ git+https://github.com/OCA/openupgradelib.git@master"
        ]
    },
```

**Comment by sbidoul** [comment]
@legalsylvain it's

```
"external_dependencies": {
        "python": [
            "openupgradelib"
        ]
    },
```

There is no way to declare the git URL in external dependencies.

**Comment by legalsylvain** [comment]
> There is no way to declare the git URL in external dependencies.

I mean, in V16 or before, there was a way to define alternative package in setup.py. There is a way to mention git url in pyproject.toml ?

**Comment by sbidoul** [comment]
> I mean, in V16 or before, there was a way to define alternative package in setup.py.

External dependencies overrides are [still possible](https://github.com/sbidoul/whool/?tab=readme-ov-file#configuration) with pyproject.toml.

> There is a way to mention git url in pyproject.toml ?

That (a git URL) was not possible with setup.py either.

The only place we can do this is in `test-requirements.txt`, IIRC.

**Comment by legalsylvain** [comment]
Thanks a lot @sbidoul

**Comment by tracencode** [comment]
Hello @legalsylvain @pedrobaeza , is there any specific date for openupgrade-17.0 release?

**Comment by rvalyi** [comment]
> Hello @legalsylvain @pedrobaeza , is there any specific date for openupgrade-17.0 release?

Unfortunately not really. I commented about that here https://github.com/OCA/OpenUpgrade/discussions/4332#discussioncomment-8698613
